test/k8sT/DatapathConfiguration.go

```diff
@@ -645,9 +645,86 @@ var _ = Describe("K8sDatapathConfig", func() {
 			}, DeployCiliumOptionsAndDNS)
 			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test between nodes failed")
 		})
+
+		SkipItIf(helpers.RunsOnGKE, "With VXLAN", func() {
+			deploymentManager.DeployCilium(map[string]string{
+				"hostFirewall": "true",
+			}, DeployCiliumOptionsAndDNS)
+			testHostFirewall(kubectl)
+		})
+
+		// We need to skip this test when using kube-proxy because of #14859.
+		SkipItIf(helpers.RunsWithKubeProxy, "With native routing", func() {
+			options := map[string]string{
+				"hostFirewall":         "true",
+				"tunnel":               "disabled",
+				"autoDirectNodeRoutes": "true",
+			}
+			// We can't rely on gke.enabled because it enables
+			// per-endpoint routes which are incompatible with
+			// the host firewall.
+			if helpers.RunsOnGKE() {
+				options["gke.enabled"] = "false"
+			}
```

Comment:
I don't think GKE guarantees to allocate nodes in the same L2, so I'm not sure whether `autoDirectNodeRoutes` is effectual in that deployment (GKE already makes pod ranges routable on the underlying network by default). Maybe that setting should only be enabled in !GKE case?

Reply:
Thanks! We didn't catch this in CI because this test won't run in CI for now (`RunsWithKubeProxy` is true on GKE CI for now). I've fixed it and validated with:
```
CNI_INTEGRATION=gke KUBEPROXY=0 K8S_VERSION=1.16 KERNEL=419 ginkgo -v --focus="K8sDatapath.* Host firewall" -- -cilium.provision=false -cilium.kubeconfig=`echo ~/.kube/config` -cilium.image="docker.io/cilium/cilium" -cilium.operator-image="docker.io/cilium/operator" -cilium.hubble-relay-image="docker.io/cilium/hubble-relay" -cilium.passCLIEnvironment=true -cilium.holdEnvironment=true
```